macros/audit.sql

```diff
@@ -46,39 +52,75 @@
 {% endmacro %}
 
 
-{% macro create_audit_log_table() %}
+{% macro create_audit_log_table() -%}
 
-    create table if not exists {{ logging.get_audit_relation() }}
-    (
-       event_name       varchar(512),
-       event_timestamp  {{dbt_utils.type_timestamp()}},
-       event_schema     varchar(512),
-       event_model      varchar(512),
-       invocation_id    varchar(512)
-    )
+    {% set required_columns = [
+       ["event_name", "varchar(512)"],
+       ["event_timestamp", dbt_utils.type_timestamp()],
+       ["event_schema", "varchar(512)"],
+       ["event_model", "varchar(512)"],
+       ["event_user", "varchar(512)"],
+       ["event_target", "varchar(512)"],
+       ["event_is_full_refresh", "boolean"],
+       ["invocation_id", "varchar(512)"],
+    ] -%}
 
-{% endmacro %}
+    {% set audit_table = logging.get_audit_relation() -%}
+
+    {% set audit_table_exists = adapter.get_relation(audit_table.database, audit_table.schema, audit_table.name) -%}
+
+
+    {% if audit_table_exists -%}
+
+        {%- set columns_to_create = [] -%}
+
+        {%- set existing_columns = adapter.get_columns_in_relation(audit_table)|map(attribute='column')|list -%}
+
+        {%- for required_column in required_columns -%}
+            {%- if required_column[0] not in existing_columns -%}
```

Comment:
For Snowflake reasons:
```suggestion
            {%- if required_column[0] not in existing_columns | lower -%}
```

Reply:
Basically, Snowflake returns column  names as upper case strings. It's silly. But for that reason, `'event_name' != 'EVENT_NAME'`

Reply:
tah, didn't know that! Added the `lower `when getting existing_columns